### ISSUE TITLE
Crash at -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:].

### ISSUE THREAD

**Issue by swarandeepsran** [uxcam/uxcam-ios-sdk#4]
App getting crashed at -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]. About 100 users are getting this crash daily. It is getting crashed in case of slow internet speed.

-[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
NSInvalidArgumentException - -[__NSCFType set]: unrecognized selector sent to instance <address>

Occurs on different iOS versions
<img width="595" alt="image" src="https://github.com/uxcam/uxcam-ios-sdk/assets/107177785/a9be0195-7efe-4dea-ad2c-c8a0a55537da">

Mostly occuring on iphone
<img width="595" alt="image" src="https://github.com/uxcam/uxcam-ios-sdk/assets/107177785/445c8660-d8b9-4632-8e8c-b962ff172ff8">

Stack Trace
```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0xec678 (Missing UUID 5a6c1f41bf7032f6a1d65b894dd21362)
1  libobjc.A.dylib                0x2bc80 objc_exception_throw
2  CoreFoundation                 0x17cfdc (Missing UUID 5a6c1f41bf7032f6a1d65b894dd21362)
3  CoreFoundation                 0x31d88 (Missing UUID 5a6c1f41bf7032f6a1d65b894dd21362)
4  CoreFoundation                 0x172950 (Missing UUID 5a6c1f41bf7032f6a1d65b894dd21362)
5  UIFoundation                   0x4f28 (Missing UUID 5abb482d3c0a3832916c8d0ac2eaa7cb)
6  UIFoundation                   0x9928 (Missing UUID 5abb482d3c0a3832916c8d0ac2eaa7cb)
7  UIFoundation                   0x7884 (Missing UUID 5abb482d3c0a3832916c8d0ac2eaa7cb)
8  UIFoundation                   0xede8 (Missing UUID 5abb482d3c0a3832916c8d0ac2eaa7cb)
9  housing                        0x1b2b1a8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
10 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
11 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
12 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
13 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
14 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
15 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
16 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
17 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
18 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
19 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
20 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
21 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
22 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
23 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
24 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
25 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
26 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
27 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
28 housing                        0x1b2b2d8 -[UXViewVideoHandler drawViewHierarchy:intoContext:parentAlpha:]
29 housing                        0x1b29d9c -[UXViewVideoHandler renderSchematicFrame:keyboardFrame:]
30 housing                        0x1b29ba4 __48-[UXViewVideoHandler schematicFrameScreenImage:]_block_invoke
31 libdispatch.dylib              0x26a8 (Missing UUID c49a5fc2c87339139603004f5d0175e7)
32 libdispatch.dylib              0x4300 (Missing UUID c49a5fc2c87339139603004f5d0175e7)
33 libdispatch.dylib              0xb894 (Missing UUID c49a5fc2c87339139603004f5d0175e7)
34 libdispatch.dylib              0xc3c4 (Missing UUID c49a5fc2c87339139603004f5d0175e7)
35 libdispatch.dylib              0x17004 (Missing UUID c49a5fc2c87339139603004f5d0175e7)
36 libdispatch.dylib              0x16878 (Missing UUID c49a5fc2c87339139603004f5d0175e7)
37 libsystem_pthread.dylib        0x1964 (Missing UUID 8f9c865b29f5346badd5fde40e008c66)
38 libsystem_pthread.dylib        0x1a04 (Missing UUID 8f9c865b29f5346badd5fde40e008c66)


```

**Comment by gabrieluxcam** [comment]
Hello @swarandeepsran thanks for sharing this. If I may ask, which SDK version are you experiencing this on? If not on the latest, please upgrade to it, you can see the changelog [here](https://developer.uxcam.com/docs/changelog). 

If you're already in a recent version, would it be possible to share your current UXCam implementation code? 

Thanks in advance.

**Comment by swarandeepsran** [comment]
@gabrieluxcam  I have tried V. 3.6.10 but crashes are happening. 

Initialization code:

```
  func initializeUXCam() {
        UXCam.optIntoSchematicRecordings()
        let configuration = UXCamConfiguration(appKey: appKey)
        UXCam.start(with: configuration)
  }
```

logging event code
```
    func sendEvent( label: inout [String: Any], screenName: String, base: inout [String: Any], name: String) {
        label[SCREEN_NAME] = screenName
        base.merge(dict: label)
        if let experiment = base["experiment"] as? [String: Any?] {
            base["experiment"] = nil
            base.merge(dict: experiment)
        }
        UXCam.logEvent(name, withProperties: base)
    }
```

**Comment by gabrieluxcam** [comment]
Hey @swarandeepsran  thank you for the details, we will work on it as soon as possible, however our developers already have some comments and questions: 

Would it be possible to look at the Firebase data and see if it's happening on certain screen or multiple screens? 

Also, could you confirm if you're using `NSAttributedString` in that screen or the project? 

For now, you're using schematic mode and turning this off should disable the crash, however in order to disable it and still record sessions we need to enable native recording for you temporarily, for that, if you wish to, you can send us a message at team@uxcam.com referring to this ticket and me and we can handle it. 

For the time being, we will work on this asap and provide and update as soon as we release the fix, thanks fro bringing it up!

**Comment by swarandeepsran** [comment]
@gabrieluxcam
1. The crash is happening on mainly one ViewController (though there are a few instances of another viewcontroller). 
2. Yes, we are using `NSAttributedString` there.

**Comment by gabrieluxcam** [comment]
Hello @swarandeepsran 

Would it be possible to provide the NSAttributedString code in that ViewController? Thanks!

**Comment by gabrieluxcam** [comment]
Thanks @swarandeepsran  for the info, our developers continue working on this issue and will provide an update asap, I will make sure to let you know! Thanks

**Comment by gabrieluxcam** [comment]
Hello @swarandeepsran  this issue has now been fixed in the new iOS SDK update 3.6.11, so it is highly suggested to upgrade asap. 

I will mark this as closed but feel free to reach out if anything.
